src/App.tsx

```diff
@@ -17,18 +36,27 @@ export const App: React.FC = () => {
             <h1 className="title">Todos:</h1>
 
             <div className="block">
-              <TodoFilter />
+              <TodoFilter setTodos={setTodos} />
             </div>
 
             <div className="block">
-              <Loader />
-              <TodoList />
+              {loader ? (
+                <Loader />
+              ) : (
+                <TodoList
+                  todos={todos}
+                  currentTodo={currentTodo}
+                  setCurrentTodo={setCurrentTodo}
+                />
+              )}
             </div>
           </div>
         </div>
       </div>
 
-      <TodoModal />
+      {currentTodo?.id && (
```

Comment:
```suggestion
      {currentTodo && (
```
enough to check currentTodo